### ISSUE TITLE
splitting up in submodules

### ISSUE THREAD

**Issue by wilzbach** [stonemaster/dlang-tour#454]
Hey all contributors and reviewers to the translations (@vild @lodo1995 @trikko @WebFreak001 @nordlow @vladdeSV @jacob-carlborg @stonemaster @Dicebot @DmitryOlshansky @dmi7ry)

First of all: thanks a lot for your hard work! It's great to see so many PRs & comments popping in lately :)

As we are starting to be indirect reviewers for languages we don't speak or understand, I thought it will be a lot fairer to give _all_ of you the permission and responsibility to manage your language yourself.
Hence I already went ahead and created a couple of teams at https://github.com/orgs/dlang-tour 

I haven't discussed this @stonemaster yet,  but here are my ideas:
- every language has its own repo with the markdown content files (no worries we can fix the edit button easily)
- you are in charge of your language (and have the rights to make it beautiful or ugly)
- there will be a base repo that pulls in in all language files (for the start at least we probably will manually update the submodules or should we directly opt for automatic deployment?)
- currently Travis checks that all examples are runnable and don't exceed the maximal length -> of course this check will be configured for each sub-module
- we could add an option to the CLI tool `dlang-tour` that allows to set a language folder, s.t. it's easy to preview the rendered version locally (and run the checks)

Before we start with this transition I wanted to ask you all whether there's anything you would miss on the list or think should be done differently?

Btw a nice feature of Github is that one can use @dlang-tour/swedish, @dlang-tour/german and even @dlang-tour/all to include all involved member for an issue/PR ;-)

Last, but not least of course I would like to hear the opinion of @stonemaster on this proposal as he started this great project and without him I couldn't ping you all.

**Comment by lodo1995** [comment]
This looks like a nice idea.

> you are in charge of your language (and have the rights to make it beautiful or ugly)

Speaking about this, I'd exploit this conversation to bring up a question I was thinking.
How exactly the translation should relate to the "original"? Should it be a one to one translation (as I think everyone is doing it now) of every paragraph, with only minor rearrangements due to the language rules? Should we use more flexibility and freedom in writing the contents of the sections?

This is also relates to how translations should be updated as the english content evolves. If it is a one to one translation, every change or addition (except for spelling corrections) should be notified to every translation team, so that it is forwarded to all languages. If instead the translation is more flexible, every translation behaves as a completely different project, with a separate set of "conceptual errors" (opposite of "spelling errors") and missing things.

Each choice has its own pros and cons, so what do you think?

**Comment by stonemaster** [comment]
> Speaking about this, I'd exploit this conversation to bring up a question I was thinking.
How exactly the translation should relate to the "original"? Should it be a one to one translation (as I think everyone is doing it now) of every paragraph, with only minor rearrangements due to the language rules? Should we use more flexibility and freedom in writing the contents of the sections?

IMO a translation as closely as possible the English version. We should make sure that all languages have the same, high quality. If a translation allows us to find errors or finds a way to better describe something, this of course should progate to English and all other translations. If we let translations to much freedom and they diverge from the original we risk that we get very heterogeneous tour where you will need to read _all_ languages to get the whole idea. I don't think we should let that happen. And different translations isn't the way other languages project go as far as I can tell. For other ways to form the content I think projects in native language are the more suitable way.

**Comment by lodo1995** [comment]
> IMO a translation as closely as possible the English version. [...]

Good. Then we (every translation team) should make sure to either follow every change to the english tour and apply it to our translation or at least do periodical checks and "bulk-updates".

> And different translations isn't the way other languages project go as far as I can tell.

Well it's very difficult to find languages that provide an official tour with translations in various languages. The only one I know about is Go (which IMO made the terrible mistake of putting the translation list on the second page, but that's another story).

**Comment by wilzbach** [comment]
> Good. Then we (every translation team) should make sure to either follow every change to the english tour and apply it to our translation or at least do periodical checks and "bulk-updates".

The GoLang Tour has a dedicated mailing list for this purpose:

https://github.com/golang/tour/blob/master/TRANSLATE

However I would personally ping @dlang-tour/all whenever there's (a) a new page, (b) or a significant change has been merged or is about to be merged. 

> The only one I know about is Go

There are efforts to translate the Rust book as well, but it's not officially supported:

https://github.com/rust-kr/doc.rust-kr.org
https://github.com/jadpole/rust-book-fr
https://github.com/kgv/rust_book_ru

Are there more interactive tours about which you know?

**Comment by Dicebot** [comment]
I disagree about 1-to-1 translation. Goal of the tour is the "first 5 minute effect" improvement. It is almost impossible to get a decently looking nice to read text by doing direct translation between most languages. As long as meaning stays the same, native language correctness and aesthetics should take the priority.

**Comment by stonemaster** [comment]
> I disagree about 1-to-1 translation. Goal of the tour is the "first 5 minute effect" improvement. It is almost impossible to get a decently looking nice to read text by doing direct translation between most languages. As long as meaning stays the same, native language correctness and aesthetics should take the priority.

I totally agree with you. I didn't mean that it should be a 1-1 translation. Of course languages are different and there are ways to describe things in a language that wouldn't be possible in English for example. I just mean that the content should be more or less be the same. So if a paragraph talks about `@safe` it should follow the content and not add more (or remove) content that isn't present in the other languages. Writer's freedom should be the highest priority, as long as we talk about the same things in all languages :-)

**Comment by wilzbach** [comment]
Hey all,

just a quick update

1) With #457 the English content has been `submodularized" and the tour now understands submodules
2) You should all be member of your "language" team
3) All repos have been initialized and have force-push protection with CI check enabled. This is the same behavior as on this repo (Travis will check whether all examples are runnable)
4) The repos are

https://github.com/dlang-tour/english
https://github.com/dlang-tour/german
https://github.com/dlang-tour/italian
https://github.com/dlang-tour/russian
https://github.com/dlang-tour/swedish

5) You can directly start adding content there. For translation repos with open PRs and files merged (german, italian, swedish) the content hasn't been moved / "submodularized" yet.
6) At the moment the main website won't update itself when a PR in a language repo is merged, we are working on this & this will happen soon (this shouldn't block your work)
7) Let us know if you have any questions & we hope that this transition phase is moderately easy

**Comment by wilzbach** [comment]
I think this has been done successfully since a long time.
